### PR TITLE
Titles

### PR DESCRIPTION
**Pull request by alexdy2007**
Began analysing titles and abstracted data_extractors to separate class.

Note: changed list of cities to include all with population above 1000 not 50000 as alot of towns in titles still persisted.

### CODE REVIEW

File: src/data_extractor/data_getter.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        :return:list of towns
+        """
+        with open(cls.TOWN_DATA, 'r') as f:
+            cities = []
+            for city in f.readlines():
+                cities.append(city.lower().replace("\n", ""))
+            return cities
+
+    @classmethod
+    def get_raw_training_data(cls, header=None):
```
Comment (utkuozbulak) [inline src/data_extractor/data_getter.py]: Just curious, any reason why not read data with pandas with a single line ? pd.read_csv(file)

Comment (charlienewey) [reply src/data_extractor/data_getter.py]: :+1:

File: src/exploration/job_titles.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import os
+from src.data_extractor.data_getter import DataGetter
+import RAKE
+
+STOP_WORDS_WIHT_CITIES_PATH = "../../stop_word_with_cities.txt"
```
Comment (charlienewey) [inline src/exploration/job_titles.py]: > STOP_WORDS_WIHT_CITIES_PATH

**Review by charlienewey — COMMENTED** [review]
small fixes, but then LGTM

**Comment by utkuozbulak** [comment]
Im guessing you will update this with @arahayrabedian  when issues with titles are done ?

**Comment by alexdy2007** [comment]
Yeah, changed most of it now with @arahayrabedian . Should hopefully finish by tomorrow.

General steps

step 1) remove stop words                        
step 2) remove other stuff                        
step 3) map title to list of jobs in archive                        
step 4) map extra info to modifier vector list                        
step 5) make it binary classification

@utkuozbulak : not loading into pandas as reading into a list to remove stop words and other preprocessing things. planning on then sticking it into a vector:

**Comment by utkuozbulak** [comment]
"@utkuozbulak : not loading into pandas as reading into a list to remove stop words and other preprocessing things. planning on then sticking it into a vector"

`data = pd.read_csv('file')  # Already in main`
`specific_feature = data[['column_name']]`
`feature_as_list = pandas_vector_to_list(specific_feature )`

`def pandas_vector_to_list(pandas_df):  # Already in cleaning functions`
`  py_list = [item[0] for item in pandas_df.values.tolist()]`
`  return py_list`

This is super simple instead of reading manually, no ?

**Comment by alexdy2007** [comment]
I don't mind either way, manual or not,

**Comment by arahayrabedian** [comment]
this is all contained as part of #19 , closing, should have just used this to be honest. ma bad.

**Pull request closed without merging**